src/components/utils/src/threads/posix_thread.cc

```diff
@@ -249,6 +249,10 @@ bool Thread::start(const ThreadOptions& options) {
   return pthread_result == EOK;
 }
 
+void Thread::yield() {
+  sched_yield();
```

Comment:
Also check #ifdef BUILD_TESTS wrapper at the beginning of file. Probably it is not required now.